common/src/main/scala/com/avast/cactus/ConverterMethods.scala

```diff
@@ -0,0 +1,44 @@
+package com.avast.cactus
+
+import cats.implicits._
+
+trait ConverterMethods[A, B] {
+  self: Converter[A, B] =>
+
+  final def map[C](f: B => C): Converter[A, C] = new Converter[A, C] {
+    override def apply(fieldPath: String)(a: A): ResultOrErrors[C] = self.apply(fieldPath)(a).map(f)
+  }
+
+  final def flatMap[C](c: Converter[B, C]): Converter[A, C] = new Converter[A, C] {
+    override def apply(fieldPath: String)(a: A): ResultOrErrors[C] = self.apply(fieldPath)(a).flatMap(c.apply(fieldPath))
+  }
+
+  final def flatMap[C](c: B => ResultOrErrors[C]): Converter[A, C] = new Converter[A, C] {
```

Comment:
Why to have two overloads?

Reply:
That's a question. Why not to? :-) I wanted to give the programmer a possibility to use what is more suitable for him - either existing `Converter` or just implement the lambda in-place.

Reply:
I don't know. For some reason this seems strange to me. Actually I think that `map` and `flatMap` shoud receive the functions and the overloads with `Converter` shouldn't be there or if then they should have a different name (`compose`?) What do you think @hanny24?

Reply:
Maybe the `compose` make sense - I don't insist on current naming.

Reply:
I would either keep the `flatMap` with "regular" signature or rename those methods to something else. This might case some pretty funny errors if anyone tries to use `for` comprehension.

I would also suggest to add methods `pure` and `failure` to `Converter` companion object. That would make whole API more standard (and IMHO much nicer)